### ISSUE TITLE
Ubuntu 16.04.5 Initial run OK, subsequent runs hang

### ISSUE THREAD

**Issue by ray-jones** [vladimiry/email-securely-app#107]
The operating system in use is Ubuntu 14.04.5 LTS.  In the commands shown, where it says `$APPIMAGE`, the actual string that was typed used the full filename `email-securely-app-2.3.1-linux-x86_64.AppImage`.

The problem being reported is that when the program is executed for a second time, a number of processes are created but the GUI never appears. The problem was originally observed while using ProtonMail.  Initial configuration and use were normal, but exiting and restarting the program caused processes to appear but no GUI.

However, in the tests below, no email account was configured, yet the problem still occurred. So the problem is unrelated to any specific email account.

Only runs 12 and 13 are described below.  I have logs from runs 1 through 11 too, but they are omitted here as runs 12 and 13 reduce the situation to its simplest possible case: start the GUI, then just exit. Also, runs 1 through 11 showed that deleting the `settings.bin` file allows normal execution again.

Also in these earlier runs I did try disabling "Clear Electron's cache storage on start and logout" and it made no difference. Also in these earlier runs enabling or disabling the local store when using ProtonMail made no difference. Also in these earlier runs exiting with the Exit menu item, or exiting by doing a Log out then closing the window, made no difference. In other words, I tried a lot of things, and nothing seemed to prevent the hang on the second run (except deleting `settings.bin` which presumably makes the second run essentially equivalent to the first run).

**Run 12**

This simulates a first-time-ever run of the program.

Killed all prior processes with `pkill email-securely`. No output was seen from `pgrep email-securely`.

Did `rm -rf ~/.config/email-securely-app`.

Ran the program from the Linux shell by typing: `./$APPIMAGE`

Entered a simple master password when prompted, and checked 'Keep me signed in'. When prompted to enter account information, did not enter any but clicked on Close. Then clicked on (down arrow) Exit to cause email-securely to exit.

Apparently it exited normally. No processes remained active (no output from `pgrep email-securely`) and in the shell window where ./$APPIMAGE had been run, it had exited with no output.

**Run 13**

Ran the program from the Linux shell by typing: `./$APPIMAGE`

No GUI was seen after several minutes, and `pgrep email-securely` showed 6 active processes.  I collected data with the command `bash x4.sh >& x4-run13.log` and the log file is attached:

[x4-run13.log](https://github.com/vladimiry/email-securely-app/files/2864632/x4-run13.log)

The script `x4.sh` uses `strace` to get the current activity of each process, and saves the first 20 lines of output, with a 10-second timeout.  This is `x4.sh`:

<PRE>
pgrep email-securely | while read pid ; do
  echo :: $pid ::
  ps -p $pid -o bsdstart,bsdtime,stat,args
  sudo timeout 10s strace -p $pid |& head -20
done
</PRE>

The log file contains lines like these:

<PRE>
:: 32435 ::
 START   TIME STAT COMMAND
 01:37   0:00 S+   /tmp/.mount_email-0w70ST/email-securely-app --type=zygote --no-sandbox
strace: Process 32435 attached
ppoll([{fd=3, events=POLLIN}], 1, NULL, [], 8strace: Process 32435 detached
 <detached ...>
:: 32474 ::
 START   TIME STAT COMMAND
</PRE>

The first line contains the pid (here it's 32435). Second and third lines contain output from the `ps` command, showing: clock time MM:SS since the process started; CPU time used MM:SS; the run state of the process; and the entire command line and arguments. The rest of the lines are output from `strace -p $pid`.

More details are available if needed.

**Comment by vladimiry** [comment]
Thanks for the thorough report.

- Can you try disabling `findInPage`, `fullTextSearch`, `checkForUpdatesAndNotify`, `clearSession` and `startMinimized` flags in `config.json` by setting them all to `false` running the most recent released version?
- Can you try running packages from here https://www.dropbox.com/sh/fxdz5efvtt0yx1y/AADB_vn2G2uL4aSBJH81aesEa?dl=0&lst= with above listed options disabled (there are not yet released the most actual `deb` and `snap` packages)?

**Comment by ray-jones** [comment]
Thanks, I will follow your suggestions and provide more details soon.

**Comment by ray-jones** [comment]
OK, good news, disabling the options you listed seems to have fixed the problem, based on quick testing both before and after adding a ProtonMail account. I didn't try the dropbox packages, only the released 2.3.1 i was already using.

I will test the options individually soon and will let you know which one(s) were responsible for the problem.

**Comment by ray-jones** [comment]
Early testing indicates that only `startMinimized` needs to be disabled. I re-enabled all the other flags that I had disabled above.

Now I have been able to add a ProtonMail account with local store enabled, and it seems to be working normally (including full-text search) even after a several exits and restarts.

**Comment by vladimiry** [comment]
Enabling `startMinimized` option makes the app's window to be hidden on app start. There are only a few ways to show the hidden window:
- Clicking respective tray icon. This is a primary option. I guess your system doesn't have a tray panel at all (Gnome shell or something?) and so you simply can't make the window visible.
- Another tricky option is executing app's binary second time (while the app is running), you can bind this command to hotkey.

The reason why the app's window is visible initially is that at first start [you don't have the `setting.bin` created](https://github.com/vladimiry/email-securely-app/blob/68a4777c8e012a6c1263bec8a3eebb702255aa70/src/electron-main/window.ts#L54).

Anyway disabling `startMinimized` option seems to solves your issue, but I'd recommend enabling the tray panel.

**Comment by ray-jones** [comment]
OK, now I understand, how stupid of me! I looked and am able to find the icon in the system tray. So that was the problem all along and not a bug at all.

I do have a convoluted explanation, though. The first time I ran the Appimage binary, it added itself to the system menus, but I could not find that entry (it's in the Office section where I probably never looked), so I manually added a start-up icon and used some icon other than the official one. Then I forgot I had done so and didn't notice a different icon appearing in the system tray.

Thanks for the suggestions, which did finally work for me.
